### DISCUSSION TITLE
with vue3.4 + router4 or above useRoute() or useRouter() will return undefined

### DISCUSSION THREAD

**Discussion by jesseice** [vuejs/router#2124] [Help and Questions] [answered]
### Reproduction

empty

### Steps to reproduce the bug

const route = useRoute();
route // undefined

### Expected behavior

return the current routing information

### Actual behavior

return undefined

**Comment by posva** [comment, marked as answer]
`useRoute()` must be used within script setup. It's only undefined when called in the wrong places. BTW, use discussions or the discord chat to ask questions, thanks!
